### PR TITLE
Fixes an issue with flashers and mounted igniters, unanchored flashers will not appear as on

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do
Fixes a "STAT OVERFLOW"(get it stack overflow?) with flashers and mounted igniters, moving their overlays to update_overlays(), and make them appear disabled when unanchored. Adds an emissive for active portable flashers, which it was previously missing.

## Why It's Good For The Game
Having stat = "16777213" is not good, consistency with updating overlays, and having visual differences for objects is good

## Images of changes
An unanchored portable flasher next to an anchored one, in the dark.
image

## Changelog
:cl:
tweak: Unanchored Portable flashers will show they are disabled
fix: Flashers will now show when they have no power
fix: Mounted igniters will now show when they have no power
imageadd: Portable flashers now have an emissive texture
/:cl:

### CODE REVIEW

File: code/game/machinery/flasher.dm [hunk 1]
```diff
@@ -125,10 +128,9 @@
 	anchored = !anchored
 	if(anchored)
 		WRENCH_ANCHOR_MESSAGE
-		overlays.Cut()
 	else
 		WRENCH_UNANCHOR_MESSAGE
-		overlays += "[base_state]-s"
+	update_icon()
```
Comment (hal9000PR) [inline code/game/machinery/flasher.dm:133]:
```suggestion
	update_icon(UPDATE_OVERLAYS)
```
dont you only want to update the overlays?

Comment (Contrabang) [reply code/game/machinery/flasher.dm:133]: For unanchoring it, it also changes the icon_state along with the overlay so I don't think I should only be overlays

**Comment by Contrabang** [comment]
I think I have the emissive light layer showing when there's no power currently, not sure if that's an issue. Won't be able to test it/change it for a few hours

File: code/game/machinery/flasher.dm [hunk 2]
```diff
@@ -34,30 +34,32 @@
...
-		stat |= ~NOPOWER
+		stat |= NOPOWER
 		set_light(0)
 	update_icon()
 
 /obj/machinery/flasher/update_icon_state()
 	. = ..()
 
-	if(stat & NOPOWER)
+	if((stat & NOPOWER) || anchored == FALSE)
```
Comment (SteelSlayer) [inline code/game/machinery/flasher.dm:48]:
```suggestion
	if((stat & NOPOWER) || !anchored)
```

File: code/game/machinery/flasher.dm [hunk 3]
```diff
@@ -34,30 +34,32 @@
...
 	. = ..()
 	underlays.Cut()
-
+	overlays.Cut()
 	if(stat & NOPOWER)
 		return
 
-	underlays += emissive_appearance(icon, "[base_state]_lightmask")
+	if(anchored)
+		overlays += "[base_state]-s"
```
Comment (SteelSlayer) [inline code/game/machinery/flasher.dm:61]:
```suggestion
		. += "[base_state]-s"
```

File: code/game/machinery/flasher.dm [hunk 4]
```diff
@@ -34,30 +34,32 @@
...
+	if((stat & NOPOWER) || anchored == FALSE)
 		icon_state = "[base_state]1-p"
 	else
 		icon_state = "[base_state]1"
 
 /obj/machinery/flasher/update_overlays()
 	. = ..()
 	underlays.Cut()
-
+	overlays.Cut()
```
Comment (SteelSlayer) [inline code/game/machinery/flasher.dm:56]:
```suggestion
	cut_overlays()
```

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**